diff --git a/dist/vite/plugin.js b/dist/vite/plugin.js
index 3b959dc6a3dd43d0ced9a3c4e3cb0bb000a0afdf..93eda16f23e5dbb0bbf0cb00c482e66c23879e69 100644
--- a/dist/vite/plugin.js
+++ b/dist/vite/plugin.js
@@ -1172,7 +1172,7 @@ async function handlePrerender(viteConfig, reactRouterConfig, serverBuildDirecto
     await prerenderRoute(handler, path, clientBuildDirectory, reactRouterConfig, viteConfig, data ? {
       headers: {
         ...headers,
-        "X-React-Router-Prerender-Data": data
+        "X-React-Router-Prerender-Data": encodeURI(data)
       }
     } : {
       headers
@@ -1226,17 +1226,26 @@ async function prerenderRoute(handler, prerenderPath, clientBuildDirectory, reac
   let normalizedPath = `${reactRouterConfig.basename}${prerenderPath}/`.replace(/\/\/+/g, "/");
   let request = new Request(`http://localhost${normalizedPath}`, requestInit);
   let response = await handler(request);
-  let html = await response.text();
-  validatePrerenderedResponse(response, html, "Prerender", normalizedPath);
-  if (!reactRouterConfig.ssr) {
-    validatePrerenderedHtml(html, "Prerender");
-  }
-  // Write out the HTML file
   let outdir = path__namespace.relative(process.cwd(), clientBuildDirectory);
-  let outfile = path__namespace.join(outdir, ...normalizedPath.split("/"), "index.html");
-  await fse__namespace.ensureDir(path__namespace.dirname(outfile));
-  await fse__namespace.outputFile(outfile, html);
-  viteConfig.logger.info(`Prerender: Generated ${colors__default["default"].bold(outfile)}`);
+
+  if (response.headers.get('Content-Type') === 'text/html') {
+    let html = await response.text();
+    validatePrerenderedResponse(response, html, "Prerender", normalizedPath);
+    if (!reactRouterConfig.ssr) {
+      validatePrerenderedHtml(html, "Prerender");
+    }
+    // Write out the HTML file
+    let outfile = path__namespace.join(outdir, ...normalizedPath.split("/"), "index.html");
+    await fse__namespace.ensureDir(path__namespace.dirname(outfile));
+    await fse__namespace.outputFile(outfile, html);
+    viteConfig.logger.info(`Prerender: Generated ${colors__default["default"].bold(outfile)}`);
+  } else {
+    let bytes = await response.bytes();
+    let outfile = path__namespace.join(outdir, ...normalizedPath.split("/"));
+    await fse__namespace.ensureDir(path__namespace.dirname(outfile));
+    await fse__namespace.outputFile(outfile, bytes);
+    viteConfig.logger.info(`Prerender: Generated ${colors__default["default"].bold(outfile)}`);
+  }
 }
 async function prerenderManifest(build, clientBuildDirectory, reactRouterConfig, viteConfig) {
   let normalizedPath = `${reactRouterConfig.basename}/__manifest`.replace(/\/\/+/g, "/");
