diff --git a/dist/index.js b/dist/index.js
index 8911b1e2ccd7bb04bf010595b0627d34d72b6ab2..15afff82dc7845e8a0e8b5e8ce723dad90c1be89 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -8720,7 +8720,7 @@ function createStaticHandlerDataRoutes(manifest, future, parentId = "", routesBy
             "X-React-Router-Prerender-Data"
           );
           invariant3(encoded, "Missing prerendered data for route");
-          let uint8array = new TextEncoder().encode(encoded);
+          let uint8array = new TextEncoder().encode(decodeURI(encoded));
           let stream = new ReadableStream({
             start(controller) {
               controller.enqueue(uint8array);
